src/ORM/Connect/PDOConnector.php

```diff
@@ -557,8 +563,11 @@ public function transactionRollback($savepoint = null)
             }
         }
 
+        // Note: $this->inTransaction may not match the 'in-transaction' state in PDO
         $this->inTransaction = false;
-        return $this->pdoConnection->rollBack();
+        if ($this->pdoConnection->inTransaction()) {
+            $this->pdoConnection->rollBack();
```

Comment:
```suggestion
            return $this->pdoConnection->rollBack();
```
Returns a bool: https://www.php.net/manual/en/pdo.rollback.php

Reply:
Updated